src/namicorn.test.ts

```diff
@@ -9,192 +9,229 @@ const MAINNET_URL = 'https://mainnet.infura.io';
+          // .log(console.log)
+          .get(REQUEST)
+          .reply(200, RESPONSE);
+      }
+    }
+  });
 };
 
 beforeEach(() => {
-	nock.cleanAll();
+  nock.cleanAll();
 });
 
 afterEach(() => {
-	if (!nock.isDone()) {
-		nock.cleanAll();
-	}
+  if (!nock.isDone()) {
+    nock.cleanAll();
+  }
 });
 
 it('should work', async () => {
-	nock.cleanAll();
-	const testName = 'should work';
-	mockAPICalls('UD_API', testName, DEFAULT_URL);
-	const namicorn = new Namicorn();
-	const result = await namicorn.address('cofounding.zil', 'eth');
-	expect(result).toEqual('0xaa91734f90795e80751c96e682a321bb3c1a4186');
+  nock.cleanAll();
+  const testName = 'should work';
+  mockAPICalls('UD_API', testName, DEFAULT_URL);
+  const namicorn = new Namicorn();
+  const result = await namicorn.address('cofounding.zil', 'eth');
+  expect(result).toEqual('0xaa91734f90795e80751c96e682a321bb3c1a4186');
 });
 
 it('resolves .eth name using blockchain', async () => {
-	const testName = 'resolves .eth name using blockchain';
-	mockAPICalls('ENS', testName, MAINNET_URL);
-
-	const namicorn = new Namicorn({
-		blockchain: { ens: MAINNET_URL },
-	});
-	var result = await namicorn.address('matthewgould.eth', 'ETH');
-	expect(result).toEqual('0x714ef33943d925731FBB89C99aF5780D888bD106');
+  const testName = 'resolves .eth name using blockchain';
+  mockAPICalls('ENS', testName, MAINNET_URL);
+
+  const namicorn = new Namicorn({
+    blockchain: { ens: MAINNET_URL },
+  });
+  var result = await namicorn.address('matthewgould.eth', 'ETH');
+  expect(result).toEqual('0x714ef33943d925731FBB89C99aF5780D888bD106');
 });
 
 it('reverses address to ENS domain', async () => {
-	// const testName = 'reverses address to ENS domain';
-
-	// mockAPICalls('ENS', testName, MAINNET_URL);
-	const ens = new Ens(MAINNET_URL);
-	const eye = jest.spyOn(ens, '_resolverCallToName').mockImplementation(() => 'adrian.argent.xyz');
-	const secondEye = jest.spyOn(ens, '_getResolver').mockImplementation(() => '0xDa1756Bb923Af5d1a05E277CB1E54f1D0A127890');
-	const result = await ens.reverse(
-		'0xb0E7a465D255aE83eb7F8a50504F3867B945164C',
-		'ETH',
-	);
-	expect(eye).toHaveBeenCalled();
-	expect(secondEye).toHaveBeenCalled();
-	expect(result).toEqual('adrian.argent.xyz');
-	eye.mockRestore();
-	secondEye.mockRestore();
+  // const testName = 'reverses address to ENS domain';
+
+  // mockAPICalls('ENS', testName, MAINNET_URL);
+  const ens = new Ens(MAINNET_URL);
+  const eye = jest
+    .spyOn(ens, '_resolverCallToName')
+    .mockImplementation(() => 'adrian.argent.xyz');
+  const secondEye = jest
+    .spyOn(ens, '_getResolver')
+    .mockImplementation(() => '0xDa1756Bb923Af5d1a05E277CB1E54f1D0A127890');
+  const result = await ens.reverse(
+    '0xb0E7a465D255aE83eb7F8a50504F3867B945164C',
+    'ETH',
+  );
+  expect(eye).toHaveBeenCalled();
+  expect(secondEye).toHaveBeenCalled();
+  expect(result).toEqual('adrian.argent.xyz');
+  eye.mockRestore();
+  secondEye.mockRestore();
```

Comment:
Suppose that any `expect` has failed. In this case `mockRestore` will never be called. Are you sure it is needed in the first place? If it is, what is the workaround to ensure it is always called?